lib/galaxy/managers/hdas.py

```diff
@@ -138,13 +138,20 @@ def purge(self, hda, flush=True):
         Purge this HDA and the dataset underlying it.
         """
         user = hda.history.user or None
+        model.StorageMedia.refresh_all_media_credentials(hda.dataset.active_storage_media_associations, self.app.authnz_manager)
         quota_amount_reduction = 0
         if user:
             quota_amount_reduction = hda.quota_amount(user)
         super(HDAManager, self).purge(hda, flush=flush)
         # decrease the user's space used
         if quota_amount_reduction:
-            user.adjust_total_disk_usage(-quota_amount_reduction)
+            if len(hda.dataset.active_storage_media_associations) == 0:
+                user.adjust_total_disk_usage(-quota_amount_reduction)
+            else:
+                for assoc in hda.dataset.active_storage_media_associations:
```

Comment:
Why is it being subtracted from all of them?

Reply:
It is reduced from all the storage media where the dataset was stored, because when purging a dataset, it is removed from all the media where it was stored.